### PR TITLE
Allow ThreadTrace to save to a file

### PR DESCRIPTION
**Pull request by outofculture**
This small feature enhancement came out of a debug session. It saves the file relative to the working directory, and either way the output format has slightly more readable use whitespace.

### CODE REVIEW

**Comment by j9ac9k** [comment]
I'm good with this PR but this screams for use of the logging module; but I know that's a whole separate effort.

File: pyqtgraph/debug.py [hunk 1]
```diff
@@ -1151,12 +1152,21 @@ def start(self, interval=None):
         self.thread.start()
 
     def run(self):
+        iter = 0
+        if self.logFile is None:
+            printFile = sys.stdout
+        else:
+            printFile = open(self.logFile, 'w', encoding='utf8')
```
Comment (j9ac9k) [inline pyqtgraph/debug.py:1159]: We open the file, but it is never closed.

**Comment by outofculture** [comment]
> I'm good with this PR but this screams for use of the logging module; but I know that's a whole separate effort.

Yeah, being able to have all the logging configured to go to any or all of the log-ui, a file, stdout or stderr would be nice.

File: pyqtgraph/debug.py [hunk 2]
```diff
@@ -1151,35 +1155,51 @@ def start(self, interval=None):
...
-            with self.lock:
-                if self._stop is True:
-                    return
-                    
-            print("\n=============  THREAD FRAMES:  ================")
-            for id, frame in sys._current_frames().items():
-                if id == threading.current_thread().ident:
-                    continue
+        iter = 0
+        if self.logFile is None:
```
Comment (j9ac9k) [inline pyqtgraph/debug.py:1159]: sys.stdout behaves like a file object, so we should be able to do something like....

```python
printFile = sys.stdout if self.logFile is None else self.logFile
```

we can nick all the `if self.logFile` checks and then use `printFile.write` everywhere.

**Comment by j9ac9k** [comment]
was reading the python docs and looks like we can treat `std.stdout` as a FileObject; which could simplify a lot of this code (and make the static code checker happy too :D )... is there some example that you happen to have on hand that I could test this out with?

**Comment by outofculture** [comment]
I've been using it in ACQ4, which is nothing at all like a minimal example. Here's something like a reasonable test, though:

```python
from threading import Thread
from time import sleep

from pyqtgraph.debug import ThreadTrace


def sleep_a_little():
    for _ in range(3):
        sleep(1)


thread = Thread(target=sleep_a_little, daemon=True)

trace = ThreadTrace(interval=1, logFile="tt.log")
thread.start()
sleep(5)
trace.stop()
thread.join()
```

However, this script fails to put any text into the `tt.log` file, like I would expect it to, so I think I broke it.

**Comment by j9ac9k** [comment]
if you're actively using this in ACQ4, and have it more or less working there as is; would you be willing to try and run a different branch of pyqtgraph on there? and see if it blows up.

tl;dr I think we can ditch both print statements and `if logfile` checks everywhere here and leverage the FileObject interface to `sys.stdout`  ... I don't mind trying to make that work if you can run within ACQ4 and make sure it works as intended.

**Comment by j9ac9k** [comment]
that false positive on the static code checker is irking me... I'll mess w/ it on my branch to see if I can call `close` in a way that it convinces codeql that it will actually close the file reliably.
